qiskit/algorithms/minimum_eigen_solvers/qaoa/qaoa_circuit.py

```diff
@@ -0,0 +1,209 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Global X phases and parameterized problem hamiltonian."""
+
+from typing import Optional, Union, cast, Set
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter, ParameterVector
+from qiskit.circuit.library import BlueprintCircuit
+from qiskit.opflow import OperatorBase, I, X, CircuitStateFn, H, CircuitOp, EvolutionFactory
+
+
+class QAOACircuit(BlueprintCircuit):
+    """Global X phases and parameterized problem hamiltonian."""
+
+    def __init__(self,
+                 cost_operator: OperatorBase,
+                 param_p: int,
```

Comment:
Since `p` is unambiguos from paper etc is it really needed to pad it out to param_p. Since it is a parameter is seems like stating the obvious.

Reply:
`p` is not a valid variable from the pylint perspective and not a great name from my point of view also. To reduce the number of `pylint: disable...` I decided to call it `param_p`. I can rename to `p`, not a problem.

Reply:
in the other circuits it is called `reps` for repetitions (e.g. here https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/circuit/library/n_local/real_amplitudes.py) so let's follow same naming conventions here as well.

Reply:
Ok, let's go with `reps`.